consensus/src/dag/types.rs

```diff
@@ -83,12 +94,48 @@ impl NodeMetadata {
     }
 }
 
+#[derive(Clone, Serialize, Deserialize, CryptoHasher, Debug)]
+pub struct NodeCertificateMetadata {
```

Comment:
I'd just call it NodeCertificate, like quorum certificate, it can contain some metadata without metadata in the name 😄